client/tests/shared/utils.test.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  applyUserAssumptionsOverDefaults,
+  applyUserCostInputsOverDefaults,
+  getRestorationYearlyBreakdown,
+  transformAssumptionsData,
+} from "@shared/lib/utils";
+
+describe("shared/lib/utils", () => {
+  describe("transformAssumptionsData", () => {
+    const mockAssumptions = {
+      valid: [
+        { name: "Baseline reassessment frequency", value: "5" },
+        { name: "Buffer", value: "0.2" },
+        { name: "Carbon price increase", value: "0.03" },
+      ],
+      invalid: [
+        { name: "Invalid name", value: "10" },
+        { name: "Buffer", value: undefined },
+      ],
+    } as const;
+
+    it("should return empty object for empty input", () => {
+      const result = transformAssumptionsData([]);
+      expect(result).toEqual({});
+    });
+
+    it("should transform valid assumptions to correct format", () => {
+      const result = transformAssumptionsData([...mockAssumptions.valid]);
+
+      expect(result).toEqual({
+        baselineReassessmentFrequency: 5,
+        buffer: 0.2,
+        carbonPriceIncrease: 0.03,
+      });
+    });
+
+    it("should ignore invalid assumption names and undefined values", () => {
+      const result = transformAssumptionsData([
+        ...mockAssumptions.valid,
+        ...mockAssumptions.invalid,
```

Comment:
Same for this, atomically might make sense, but it should not happen.

What do you think we should do?